lib/generator.js

```diff
@@ -975,6 +993,9 @@
                 clipToDocumentBounds: settings.clipToDocumentBounds || false,
                 maxDimension: settings.maxDimension || 10000
             };
+        if (settings.hasOwnProperty("thread")) {
```

Comment:
is there a *problem* with allowing params.thread = false if is not supplied in the `settings` object?  If not, it seems weird to have thing dangling down here.